snapcraft/plugins/rust.py

```diff
@@ -76,9 +77,6 @@ def get_build_properties(cls):
     def __init__(self, name, options, project):
         super().__init__(name, options, project)
         self.build_packages.extend([
```

Comment:
I attempted to removed redundant packages here. Unfortunately our integration tests are not 100% trustworthy because **gcc** will implicitly be present on CI I verified this again in a LXD container and gcc is needed, the other packages aren't - at least for the test packages in snapcraft, if there's others we need to add new tests.